### PR TITLE
Consolidate workspaces into one

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously there was a `core`, `trusted`, and an `untrusted` workspace.
These have now been merged into one workspace.

The `trts` and `panic` crates have been excluded from testing since
their tests can not be run on the host system.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #21** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/21" target="_blank">Graphite</a> 
    * **PR #22** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/22" target="_blank">Graphite</a> 
      * **PR #27** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/27" target="_blank">Graphite</a> 
        * **PR #28** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/28" target="_blank">Graphite</a> 
        * **PR #32** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/32" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/32?utm_source=stack-comment).

**Review by jcape — APPROVED** [review]

**Review by NotGyro — APPROVED** [review]
Looks good, tests passing remotely as well as locally. I love the 10:1 deleted lines to added lines going on here, hah - what a reduction in complexity! Approved

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/32?utm_source=stack-comment).


Job Reference: UxB55iIz8cPF3CQr9ryU

**Pull request merged**
